src/main/io/serial.h

```diff
@@ -91,7 +91,8 @@ typedef enum {
     SERIAL_PORT_USB_VCP = 20,
     SERIAL_PORT_SOFTSERIAL1 = 30,
     SERIAL_PORT_SOFTSERIAL2,
-    SERIAL_PORT_IDENTIFIER_MAX = SERIAL_PORT_SOFTSERIAL2
+    SERIAL_PORT_IDENTIFIER_MAX = SERIAL_PORT_SOFTSERIAL2,
+    SERIAL_PORT_ALL = 255,
```

Comment:
@etracer65: I just realised that this is overflowing the variable in places where it is specified as 8 bit *signed* integer, like in parameter groups.

Reply:
Sorry about that. I guess it needs to be 127 or maybe some other negative value. Would it really matter though? The "ALL" value is only used programmatically and can't be stored in the setting.

I'm away from my computer for about the next 3 hours so feel free to correct if you can.

Reply:
The CI build for 4.2.7 locks up on F411 right after flashing (https://github.com/betaflight/betaflight/issues/10549, F405 and F722 are ok), and I bisected it to this commit, so I suspect that this is the culprit (even though I don't know how exactly).
Will do some more testing and then propose a fix.
For now I've replaced the F411 hex file in the 4.2.7 release with one that has #10490 and #10489 removed to stop people from getting stuck.

Reply:
Ok. Still mobile but was thinking if the struct had the enum type for the element then maybe it's implicitly converting to an int16 corrupting the PG.

Reply:
I confirm that a F411 FC (MATEKF411RX target, TinyHawk2) has lost all serial after upgrading to 4.2.7 from source (I have soldered a flash chip onto it for blackbox and had to add support for it, hence building from source). The one from the releases page worked well.

Reply:
PR incoming in a little bit. The problem is definitely related to the `serialPortIdentifier_e` enumeration apparently getting implicitly converted to `int16`. Either way having 255 as an enumeration is bad as everything that uses it is all `int8`. My earlier supposition about the PG getting corrupted because it used the `serialPortIdentifier_e`was incorrect as the structure uses `int8_t`. So it's good news that the PG wasn't getting misaligned/corrupted. So I'm guessing that the overflow/bit-flipping by stuffing the int16 into int8 is causing the serial code to wedge somewhere. Tested by switching `SERIAL_PORT_ALL` to -128 and tested on F411 in both upgrades from previous versions without full-chip erase (preserving the PG contents) as well as clean 4.2.7 flashes and all seems to work. Will include 4.2.7 test hexes on the PR as well.

Reply:
PR to fix the problem is submitted: https://github.com/betaflight/betaflight/pull/10550

Reply:
Funnily enough I can not get this to wedge on a different board (MATEKF411RX wedges, MATEKF411 doesn't, even if I use the MATEKF411RX configuration).

Reply:
I did my testing on a real MATEKF411RX board with the fix and all was good. I was actually able to get a CLRACINGF7 to wedge with the original code when doing an upgrade from 4.2.5 to the release 4.2.7. Well it didn't actually wedge as the status LEDs were still showing proper activity, but serial and VCP weren't working. But then doing a full-chip erase and clean install of 4.2.7 worked fine so the pattern is hard to figure out. There's definitely from weird behavior when the enumeration becomes int16 and then overflows the int8 variables. The original code was definitely tested on both F405 and F722 and never exposed the problem however.

Reply:
I think I've found the reason for the seemingly erratic behaviour - the wedging seems to be dependent on the previously present configuration - flashing with 'Full Chip Erase' got my MATEKF411RX back to work even with the problematic build. This makes sense as `serialPortIdentifier_e` is indeed used in a parameter group (`serialConfig`). Fix incoming.